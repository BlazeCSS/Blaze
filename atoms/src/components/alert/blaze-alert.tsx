import { Component, Method, Prop, State } from '@stencil/core';

@Component({
  tag: 'blaze-alert'
})
export class Alert {

  @Prop() type: string;
  @Prop() dismissible: boolean = false;
  @Prop() open: boolean = false;
  @State() _isOpen: boolean = false;

  @Method()
  close() {
    this._isOpen = false;
  }

  @Method()
  show() {
    this._isOpen = true;
  }

  @Method()
  isOpen() {
    return this._isOpen;
  }

  componentWillLoad() {
    this._isOpen = this.open;
  }

  render() {
    const isOpenClass = !this._isOpen ? 'u-display-none' : '';

    return (
      <div class={`c-text c-alert c-alert--${this.type} ${isOpenClass}`}>
        {
          this.dismissible &&
          <button class="c-button c-button--close" onClick={() => this.close()}>
            &times;
          </button>
        }
        <slot />
      </div>
    );
  }
}
