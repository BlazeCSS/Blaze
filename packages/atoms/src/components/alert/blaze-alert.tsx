import { Component, Method, Prop, State } from '@stencil/core';

@Component({
  tag: 'blaze-alert'
})
export class Alert {

  @Prop() type: string = '';
  @Prop() dismissible: boolean = false;
  @Prop() open: boolean = false;
  @State() _isOpen: boolean = false;

  @Method()
  close() {
    this._isOpen = false;
  }

  @Method()
  show() {
    this._isOpen = true;
  }

  @Method()
  isOpen() {
    return this._isOpen;
  }

  componentWillLoad() {
    this._isOpen = this.open;
  }

  render() {
    const isOpenClass = !this._isOpen ? 'u-display-none' : '';
    const typeClass = this.type ? `c-alert--${this.type}` : '';

    return (
      <div role="alert" hidden={!this._isOpen} class={`c-alert ${typeClass} ${isOpenClass}`}>
        {
          this.dismissible &&
          <button class="c-button c-button--close" onClick={() => this.close()}>
            &times;
          </button>
        }
        <slot />
      </div>
    );
  }
}
