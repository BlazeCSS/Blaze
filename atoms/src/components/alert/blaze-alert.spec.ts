import { flush, render } from '@stencil/core/testing';
import { Alert } from './blaze-alert';

describe('Alert', () => {
  it('should build', () => {
    expect(new Alert()).toBeTruthy();
  });

  let element;

  it('should work without parameters', async () => {
    element = await render({
      components: [Alert],
      html: '<blaze-alert>test this!</blaze-alert>'
    });
    await flush(element);

    expect(element).toMatchSnapshot();
  });

  it('should work with type set', async () => {
    element = await render({
      components: [Alert],
      html: '<blaze-alert type="brand">test this!</blaze-alert>'
    });
    await flush(element);

    expect(element).toMatchSnapshot();
  });

  it('renders nothing when closed', async () => {
    element = await render({
      components: [Alert],
      html: '<blaze-alert type="brand">test this!</blaze-alert>'
    });
    await flush(element);

    expect(element).toMatchSnapshot();

    element.close();
    await flush(element);

    expect(element).toMatchSnapshot();
  });

  it('renders everything when opened', async () => {
    element = await render({
      components: [Alert],
      html: '<blaze-alert type="brand">test this!</blaze-alert>'
    });
    await flush(element);

    element.close();
    await flush(element);

    element.show();
    await flush(element);

    expect(element).toMatchSnapshot();
  });
});
