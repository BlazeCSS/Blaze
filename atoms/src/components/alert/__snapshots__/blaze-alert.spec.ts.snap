// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Alert renders everything when opened 1`] = `
<blaze-alert
  class="hydrated"
  type="brand"
>
  <div
    class="c-text c-alert c-alert--brand"
  >
    
    <!---->
    test this!
  </div>
</blaze-alert>
`;

exports[`Alert renders nothing when closed 1`] = `
<blaze-alert
  class="hydrated"
  type="brand"
>
  <div
    class="c-text c-alert c-alert--brand u-display-none"
  >
    
    <!---->
    test this!
  </div>
</blaze-alert>
`;

exports[`Alert renders nothing when closed 2`] = `
<blaze-alert
  class="hydrated"
  type="brand"
>
  <div
    class="c-text c-alert c-alert--brand u-display-none"
  >
    
    <!---->
    test this!
  </div>
</blaze-alert>
`;

exports[`Alert should work with type set 1`] = `
<blaze-alert
  class="hydrated"
  type="brand"
>
  <div
    class="c-text c-alert c-alert--brand u-display-none"
  >
    
    <!---->
    test this!
  </div>
</blaze-alert>
`;

exports[`Alert should work without parameters 1`] = `
<blaze-alert
  class="hydrated"
>
  <div
    class="c-text c-alert c-alert--undefined u-display-none"
  >
    
    <!---->
    test this!
  </div>
</blaze-alert>
`;
